crates/chain/src/local_chain.rs

```diff
@@ -1,14 +1,116 @@
+//! The [`LocalChain`] is a local implementation of [`ChainOracle`].
+
 use core::convert::Infallible;
 
-use alloc::collections::{BTreeMap, BTreeSet};
+use crate::collections::BTreeMap;
+use crate::{BlockId, ChainOracle};
+use alloc::{sync::Arc, vec::Vec};
 use bitcoin::BlockHash;
 
-use crate::{BlockId, ChainOracle};
+/// A structure that represents changes to [`LocalChain`].
+pub type ChangeSet = BTreeMap<u32, Option<BlockHash>>;
+
+/// Represents a block of [`LocalChain`].
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct CheckPoint {
+    /// Block height.
+    pub height: u32,
+    /// Block hash.
+    pub hash: BlockHash,
```

Comment:
Why not just hacve `block_id` here instead of these two fields.

Reply:
No specific reason, just that it seems more convenient this way if we are exposing `CheckPoint`. I.e. you can get blockhash by `cp.hash` instead of `cp.block.hash`.

Reply:
Changed.